ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/BrokerCompatibilityCheck.java

```diff
@@ -0,0 +1,281 @@
+                                   final List<MetricsReporter> reporters,
+                                   final Time time) {
+    this.config = config;
+    this.kafkaClient = kafkaClient;
+    this.reporters = reporters;
+    this.time = time;
+  }
+
+
+  public static BrokerCompatibilityCheck create(final Config config) {
+    final Time time = new SystemTime();
+    final ChannelBuilder channelBuilder = ClientUtils.createChannelBuilder(config);
+    final LogContext logContext = new LogContext("[BrokerCompatibilityCheck] ");
+
+    final Map<String, String> metricTags = new LinkedHashMap<>();
+    final String clientId = config.getString(StreamsConfig.CLIENT_ID_CONFIG);
+    metricTags.put("client-id", clientId);
+
+    final Metadata metadata = new Metadata(config.getLong(StreamsConfig.RETRY_BACKOFF_MS_CONFIG),
+        config.getLong(StreamsConfig.METADATA_MAX_AGE_CONFIG),
+        false);
+
+    final List<InetSocketAddress> addresses = ClientUtils.parseAndValidateAddresses(config.getList(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG));
+    metadata.update(Cluster.bootstrap(addresses), Collections.<String>emptySet(), time.milliseconds());
+
+    final MetricConfig metricConfig = new MetricConfig().samples(config.getInt(CommonClientConfigs.METRICS_NUM_SAMPLES_CONFIG))
+        .timeWindow(config.getLong(CommonClientConfigs.METRICS_SAMPLE_WINDOW_MS_CONFIG), TimeUnit.MILLISECONDS)
+        .tags(metricTags);
+    final List<MetricsReporter> reporters = config.getConfiguredInstances(
+        ProducerConfig.METRIC_REPORTER_CLASSES_CONFIG,
+        MetricsReporter.class);
+    reporters.add(new JmxReporter("kafka.admin.client"));
+    final Metrics metrics = new Metrics(metricConfig, reporters, time);
+
+    final Selector selector = new Selector(
+        config.getLong(StreamsConfig.CONNECTIONS_MAX_IDLE_MS_CONFIG),
+        metrics,
+        time,
+        "kafka-client",
+        channelBuilder,
+        logContext);
+
+    final KafkaClient kafkaClient = new NetworkClient(
+        selector,
+        metadata,
+        clientId,
+        MAX_INFLIGHT_REQUESTS, // a fixed large enough value will suffice
+        config.getLong(StreamsConfig.RECONNECT_BACKOFF_MS_CONFIG),
+        config.getLong(StreamsConfig.RECONNECT_BACKOFF_MAX_MS_CONFIG),
+        config.getInt(StreamsConfig.SEND_BUFFER_CONFIG),
+        config.getInt(StreamsConfig.RECEIVE_BUFFER_CONFIG),
+        config.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG),
+        time,
+        true,
+        new ApiVersions(),
+        logContext);
+
+    return new BrokerCompatibilityCheck(config, kafkaClient, reporters, time);
+  }
+
+  /**
+   * Check if the used brokers have version 0.10.1.x or higher.
+   *
+   * <p>Note, for <em>pre</em> 0.10.x brokers the broker version cannot be checked and the client will hang and retry
+   * until it {@link StreamsConfig#REQUEST_TIMEOUT_MS_CONFIG times out}.
+   *
+   * @throws KsqlException if brokers have version 0.10.0.x
+   */
```

Comment:
KSQL 0.2 and up need Kafka 0.11 since we need a bugfix in admin client that was merged in 0.11.
However, KSQL 0.1 will work with Kafka 0.10.

Reply:
Not it won't. Streams only works with kafka 0.10.1.x and higher

Reply:
@hjafarpour the `AdminClient` bug fix, was that client only? i.e., we should be able to run with kafka 0.10.1.x and higher brokers

Reply:
Yes, that's right @dguy.

Reply:
@dguy yes, you are right. I was thinking about the KSQL dependency, but as you said it's client side and we should be fine with 0.10.1.x in broker side.